### ISSUE TITLE
feature wish: spheres collection

### ISSUE THREAD

**Issue by nilsbecker** [vispy/vispy#1071]
Since drawing many independent Visuals in a Scene is relatively slow, it would be good to have collection Visuals available for common use cases. I would like a SpheresCollection which contains spheres of varying color, diameter and center which are updated from corresponding arrays. This could be either via properly rescaled circular point sprites or via actual Meshes. It should be possible to draw a thousand spheres.

**Comment by nilsbecker** [comment]
(i tried in vain to implement this myself: it's not easy to get up to speed with opengl in a short time)

**Comment by michaelaye** [comment]
I am wondering if the Markers class is a working example for this, as it handles several thousand objects without a problem for me? I'm basically drawing several thousands of them around a sphere and it works fine. I totally agree that something like this would be helpful.

**Comment by rougier** [comment]
The [molecule example](https://github.com/vispy/experimental/blob/master/nr_gloo/molecule.py) uses a marker collection using disc and a special shader that makes look them as sphere. It's quite efficient since it is only 2 triangles / sphere. Depending on your exact need, that might just do the work.

**Comment by nilsbecker** [comment]
yes, that example seems to work well. however, it cannot easily be adapted to be used within the scene framework; the pointsize calculation used there would need to be built into some new Visual, so it can be combined with predefined cameras etc.

**Comment by campagnola** [comment]
@rougier, is there a reason for using triangles instead of points? I seem to recall coming to the same conclusion, but I don't remember why..

**Comment by rougier** [comment]
@campagnola Yes, GL_POINT_SIZE is limited on some architecture (64 pixels at most for some).
But now you ask the question, I think we use point in vispy.
